app/vminsert/influx/request_handler.go

```diff
@@ -105,9 +106,15 @@ func insertRows(at *auth.Token, db string, rows []parser.Row, mayOverrideAccount
 			ic.Labels = ic.Labels[:len(ic.Labels)-1]
 			ic.AddLabel("", metricGroup)
 			ic.MetricNameBuf = storage.MarshalMetricLabelRaw(ic.MetricNameBuf[:metricNameBufLen], placeholderLabel)
-			storageNodeIdx := ic.GetStorageNodeIdx(&atCopy, ic.Labels)
-			if err := ic.WriteDataPointExt(&atCopy, storageNodeIdx, ic.MetricNameBuf, r.Timestamp, f.Value); err != nil {
-				return err
+			storageNodeGroupIDs := ic.GetStorageNodeGroupIds(&atCopy, ic.Labels)
+			for _, storageNodeGroupID := range storageNodeGroupIDs {
+				if err := ic.WriteDataPointExt(at, storageNodeGroupID, ic.MetricNameBuf, r.Timestamp, f.Value); err != nil {
```

Comment:
This is synchronous replication, i.e. the data is written sequentially to all the replication groups. This means that the maximum data ingestion performance is limited by the slowest replication group.

For example, if one replication group is located in the remote datacenter, while another one is located in the local datacenter comparing to `vminsert`, then the ingestion performance may be limited by network latency for the remote datacenter, since `vminsert` waits for the acknowledge reply from `vmstorage`.

This also means that the data ingestion may slow down or even hang if certain replication group becomes temporarily slower or stops accepting new data.

Is this OK?